### ISSUE TITLE
text diffs on huge files are slow

### ISSUE THREAD

**Issue by d-b-w** [pytest-dev/pytest#12406]
Ok, of course they are... But usually* when a dev commits the test, the test is passing, so the dev may not notice how preposterous a diff they are inadvertently asking for.

In my case, a 5 year old test happened to be comparing text files about 2million lines long as strings, functionally:

```
    assert fh1.read() == fh2.read()
```

This was fine, until the order of some fields changed and the test started hanging in CI for hours. The right thing to do is to fix this annoying test, but I thought that it might also make sense to push a fix up to pytest.

tl;dr - `_diff_text()` already knows the verbosity level - would it make sense to truncate the length of the diff calculated in "non verbose" mode?

**Comment by nicoddemus** [comment]
@d-b-w thanks for the report!

I believe so yes. A PR in that direction would be appreciated! 👍

**Comment by d-b-w** [comment]
OK!
